ts/DataGrid/DataGrid.ts

```diff
@@ -937,6 +943,55 @@ class DataGrid {
         DataGrid.dataGrids.splice(dgIndex, 1);
     }
 
+    /**
+     * Grey out the data grid and show a loading indicator.
+     *
+     * @param message
+     * The message to display in the loading indicator.
+     */
+    public showLoading(message?: string): void {
+        // Create loading wrapper.
+        this.loadingWrapper = makeHTMLElement(
+            'div',
+            {
+                className: 'highcharts-datagrid-loading-wrapper',
+            },
+            this.contentWrapper
+        )
+
+        // Create spinner element.
+        makeHTMLElement(
+            'div',
+            {
+                className: 'highcharts-datagrid-spinner'
+            },
+            this.loadingWrapper
+        );
+
+
+        // Create loading message span element.
+        const loadingSpan = makeHTMLElement(
+            'span',
+            {
+                className: 'highcharts-datagrid-loading-message'
+            },
+            this.loadingWrapper
+        );
+
+        AST.setElementHTML(
+            loadingSpan,
+            pick(message, this.options?.lang?.loading, '')
+        );
```

Comment:
In most cases the content will not be HTML, so a micro-optimization would be to use this Util function instead:
https://github.com/highcharts/highcharts/blob/6ba3a49de9aee6490e9e41d9c634135e35da7e70/ts/DataGrid/Utils.ts#L136-L156